modules/systemd-journal/journal-reader.c

```diff
@@ -598,7 +598,22 @@ _init(LogPipe *s)
   if (!log_source_init(s))
     return FALSE;
 
-  gint res = journald_open(self->journal, SD_JOURNAL_LOCAL_ONLY);
+  gint journal_flags = SD_JOURNAL_LOCAL_ONLY;
+
+#if SYSLOG_NG_HAVE_JOURNAL_NAMESPACES
+  if (self->options->all_namespaces)
+    {
+      msg_error("all_namespaces");
+      journal_flags |= SD_JOURNAL_ALL_NAMESPACES;
+    }
+  gint res = journald_open_namespace(self->journal, NULL, journal_flags);
+#else
+  if (!self->options->all_namespaces)
```

Comment:
This is just a guess, but I think the negation was used because of the default value. Which is TRUE, regardless of the systemd version in place.
I would like to start a discussion about the expected behavior. To give a proper message to the user, a binary state variable is not enough for this warning. Since you also need a "not set" state.

My proposal: 
I am thinking about a more general solution, where we simply introduce an option, called "namespace" (as a string list), which will currently ONLY handle the "ALL" option.
If the option is not set at all, or systemd version is less than 2.45, than it will not have any effect.
This option later can be extended to support actual namespaces.

Reply:
Your guess is correct. I wanted to issue the warning only when someone tries to do something impossible on their system. The only case i could imagine was setting all_namespaces(no) on a <v245 system. But i agree, there should be a not set state.

I've tried to implement the suggested namespace() option instead off all-namespaces().

Reply:
I just wanted to start a discussion. You went straight ahead, really nice. I would wait for others opinion, but I like the direction. 

note: When squashing the commits together at the end, I think this part should be extracted into a function. So the whole "namespace string" handling logic will be encapsulated.

Reply:
I'm not very opinionated regarding the implementation, but also C is not really my strength. So I'll go with whatever is proposed and sound.

I've encapsulated the journal opening in a separate function and added a file in the news/ folder. Is there any documentation which needs to be updated? At least I couldn't find anything in this git repo.

Question: Why is the `systemd-journal()` source limited to one instance? Would it make sense to allow for more than one instance for different namespaces? Just thinking out loud here. Otherwise there might to be a lot of filtering to be done when collecting all namespaces and filtering by namespace.

Reply:
The extracted function looks good, and I think the news file is enough for the PR. (I notified the documentation team, to add the new option to the admin guide too.)

Limit to one instance: I think this is coming from the old "namespace less" version. There is no reason to add the same source multiple times. (It might also cause race conditions.)
With your modification I think it would be logical to allow multiple instances. However it have to be checked globally to make sure the namespaces are unique between sources, and only allow one "default" namespace source at once. (I would rather not widening the scope of this PR. It can be done in a separate one.)

Reply:
Agreed.